integration/tests/cook/test_basic.py

```diff
@@ -3226,6 +3226,25 @@ def test_kubernetes_checkpointing(self):
         finally:
             util.kill_jobs(self.cook_url, [job_uuid_disabled, job_uuid_enabled])
 
+    @unittest.skipUnless(util.using_kubernetes(), 'Test requires kubernetes')
+    def test_kubernetes_metadata_env_vars(self):
+        docker_image = util.docker_image()
+        container = {'type': 'docker',
+                     'docker': {'image': docker_image}}
+        default_pool = util.default_submit_pool() or util.default_pool(self.cook_url)
+        # info = util.scheduler_info(self.cook_url)
+        try:
+            # need to figure out how to match up COOK_SCHEDULER_REST_URL
+            # leader url and COOK_SCHEDULER_REST_URL can have different schemes right now
+            #command = 'bash -c \'if [[ "${COOK_POOL:-zzz}" == "' + default_pool + '" ]] && [[ "${COOK_SCHEDULER_REST_URL:-zzz}" == "' + info['leader-url'] + '" ]]; then exit 0; else exit 1; fi\''
+            command = 'bash -c \'if [[ "${COOK_POOL:-zzz}" == "' + default_pool + '" ]] && [[ "${COOK_SCHEDULER_REST_URL:-zzz}" != "zzz" ]]; then exit 0; else exit 1; fi\''
+            job_uuid, resp = util.submit_job(self.cook_url, command=command, container=container)
+            self.assertEqual(201, resp.status_code, resp.text)
+            util.wait_for_instance(self.cook_url, job_uuid, status='success')
```

Comment:
If we're going to wait for success, I think we need to add some retries to be resilient to random failures

Reply:
so going forward for new tests we want retries on any job submission? is there an example of this on a different test? can we implement retries on a test level instead, like in waiter?

Reply:
You only want the retries if you're expecting the test job to succeed (many tests don't have that expectation). Yeah, there are lots of examples, search for `max_retries=5` in the tests.

Reply:
oh, I wasn't thinking about cook retries. updated